sunpy/time/timerange.py

```diff
@@ -65,11 +67,13 @@ def __init__(self, a, b=None):
             y = b
 
         if isinstance(y, u.Quantity):
-            y = timedelta(seconds=y.to('s').value)
+            y = TimeDelta(y)
+
+        # TODO: Support datetime.timedelta?
```

Comment:
@cadair @nabobalis Should we support the case where y (ie b) is `datetime.timedelta`?